plugins/outputs/influxdb/influxdb.go

```diff
@@ -224,17 +224,19 @@ func (i *InfluxDB) Write(metrics []telegraf.Metric) error {
 
 		switch apiError := err.(type) {
 		case *DatabaseNotFoundError:
-			if !i.SkipDatabaseCreation {
-				allErrorsAreDatabaseNotFoundErrors = false
-				err := client.CreateDatabase(ctx, apiError.Database)
-				if err != nil {
-					i.Log.Errorf("When writing to [%s]: database %q not found and failed to recreate",
-						client.URL(), apiError.Database)
-				} else {
-					// try another client, if all clients fail with this error, do not return error
-					continue
-				}
+			if i.SkipDatabaseCreation {
+				continue
 			}
+
+			err := client.CreateDatabase(ctx, apiError.Database)
+			if err != nil {
+				i.Log.Errorf("When writing to [%s]: database %q not found and failed to recreate",
+					client.URL(), apiError.Database)
+			} else {
+				return errors.New("database created; retry write")
+			}
```

Comment:
Why do you return an error on success but only log on failure? This seems strange here, so I would at least expect a solid comment explaining the _why_.

Reply:
Added a comment